fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneScanAllEntriesTest.java

```diff
@@ -0,0 +1,220 @@
+                                          : LuceneIndexTestUtils.fullTextSearch(store, dataModel.index, search, false);
+
+            // Run the scan with the given query and assert the results
+            assertIndexEntryPrimaryKeyTuples(expectedResult,
+                    store.scanIndex(dataModel.index, scanBounds, null, ScanProperties.FORWARD_SCAN));
+        }
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    public void scanLargeIndexTest(boolean isGrouped) throws Exception {
+        final long seed = 6437286L;
+        final boolean isSynthetic = false;
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilder, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPrimaryKeySegmentIndexEnabled(true)
+                .setPartitionHighWatermark(10)
+                .build();
+
+        final long start = Instant.now().toEpochMilli();
+        try (FDBRecordContext context = openContext()) {
+            dataModel.saveRecords(500, start, context, 2);
+            commit(context);
+        }
+
+        // This test only uses the match-all query
+        LuceneQueryClause search = LuceneQuerySearchClause.MATCH_ALL_DOCS_QUERY;
+
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = dataModel.createOrOpenRecordStore(context);
+            LuceneScanBounds scanBounds = isGrouped
+                                          ? LuceneIndexTestValidator.groupedSortedTextSearch(store, dataModel.index, search, null, 2)
+                                          : LuceneIndexTestUtils.fullTextSearch(store, dataModel.index, search, false);
+
+            // Run the scan with the given query and assert the results
+            assertIndexEntryPrimaryKeys(l -> keyFor(2, 1001L + l), 500,
+                    store.scanIndex(dataModel.index, scanBounds, null, ScanProperties.FORWARD_SCAN));
+        }
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    public void scanWithContinuationsTest(boolean isGrouped) throws Exception {
+        final long seed = 85373450L;
+        final boolean isSynthetic = false;
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilder, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPrimaryKeySegmentIndexEnabled(true)
+                .setPartitionHighWatermark(10)
+                .build();
+
+        final long start = Instant.now().toEpochMilli();
+        try (FDBRecordContext context = openContext()) {
+            dataModel.saveRecords(500, start, context, 2);
+            commit(context);
+        }
+        // Scan properties with a limit of 100
+        final ScanProperties scanProperties = new ScanProperties(ExecuteProperties.newBuilder()
+                .setReturnedRowLimit(100)
```

Comment:
You probably don't want this to be an even multiple of the `partitionHighWatermark`